js/bundle.go

```diff
@@ -298,7 +299,24 @@ func (b *Bundle) Instantiate(
 
 // Instantiates the bundle into an existing runtime. Not public because it also messes with a bunch
 // of other things, will potentially thrash data and makes a mess in it if the operation fails.
-func (b *Bundle) instantiate(logger logrus.FieldLogger, rt *goja.Runtime, init *InitContext, vuID uint64) error {
+func (b *Bundle) instantiate(logger logrus.FieldLogger, rt *goja.Runtime, init *InitContext, vuID uint64) (err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			gojaStack := rt.CaptureCallStack(20, nil)
+
+			err = &common.InterruptError{Reason: fmt.Sprintf("a panic occurred in the init context code: %s", r)}
+			// TODO figure out how to use PanicLevel without panicing .. this might require changing
+			// the logger we use see
+			// https://github.com/sirupsen/logrus/issues/1028
+			// https://github.com/sirupsen/logrus/issues/993
+			b := new(bytes.Buffer)
+			for _, s := range gojaStack {
+				s.Write(b)
+			}
+			logger.Error("panic: ", r, "\n", string(debug.Stack()), "\nGoja stack:\n", b.String())
+		}
+	}()
```

Comment:
nit: Can we put this *just* before the `eventLoop.start` as otherwise this will catch any panic up until that point as well

Reply:
Maybe also movign this to a helper function will be ... helpful :thinking:

Reply:
Good point, a consolidation sounds about right :+1: